src/Safe/Coerce.purs

```diff
@@ -6,11 +6,18 @@ module Safe.Coerce
 import Prim.Coerce (class Coercible)
 import Unsafe.Coerce (unsafeCoerce)
 
+-- | Coerce a value of one type to a value of some other type, without changing
+-- | its runtime representation. This function behaves identically to
+-- | `unsafeCoerce` at runtime. Unlike `unsafeCoerce`, it is safe, because the
+-- | `Coercible` constraint prevents any use of this function from compiling
+-- | unless the compiler can prove that the two types have the same runtime
+-- | representation.
+-- |
 -- | One application for this function is to avoid doing work that you know is a
 -- | no-op because of newtypes. For example, if you have an `Array (Conj a)` and you
--- | want an `Array (Disj a)`, you could do `Data.Array.map (runConj >>> Disj)`, but
+-- | want an `Array (Disj a)`, you could do `Data.Array.map (un Conj >>> Disj)`, but
 -- | this performs an unnecessary traversal. `coerce` accomplishes the same
```

Comment:
Would it help to note that the unnecessary traversal in this case is `O(n)` to help show the contrast more?

Reply:
Yes, that sounds good to me